packages/react-native-web/src/modules/applyLayout/index.js

```diff
@@ -9,29 +9,73 @@
 
 import { canUseDOM } from 'fbjs/lib/ExecutionEnvironment';
 import debounce from 'debounce';
+import findNodeHandle from '../../exports/findNodeHandle';
 
 const emptyObject = {};
 const registry = {};
 
 let id = 1;
 const guid = () => `r-${id++}`;
 
+let resizeObserver;
 if (canUseDOM) {
-  const triggerAll = () => {
-    Object.keys(registry).forEach(key => {
-      const instance = registry[key];
-      instance._handleLayout();
+  if (typeof window.ResizeObserver !== 'undefined') {
+    resizeObserver = new window.ResizeObserver(entries => {
+      entries.forEach(({ target }) => {
+        target._handleLayout && target._handleLayout();
+      });
     });
-  };
+  } else {
+    if (process.env.NODE_ENV !== 'production') {
+      console.warn(
+        'onLayout relies on ResizeObserver which is not supported by your browser. ' +
+          'Please include a polyfill. https://github.com/WICG/ResizeObserver/issues/3. ' +
+          'Falling back to window.onresize.'
+      );
+    }
 
-  window.addEventListener('resize', debounce(triggerAll, 16), false);
+    const triggerAll = () => {
+      Object.keys(registry).forEach(key => {
+        const instance = registry[key];
+        instance._handleLayout();
+      });
+    };
+
+    window.addEventListener('resize', debounce(triggerAll, 16), false);
+  }
 }
 
+const observe = instance => {
+  if (resizeObserver) {
+    const node = findNodeHandle(instance);
+    node._handleLayout = debounce(instance._handleLayout.bind(instance));
+    resizeObserver.observe(node);
+  } else {
+    const id = guid();
+    instance._onLayoutId = id;
+    registry[id] = instance;
+    instance._handleLayout();
+  }
+};
+
+const unobserve = instance => {
+  if (resizeObserver) {
+    const node = findNodeHandle(instance);
+    delete node._handleLayout;
+    resizeObserver.unobserve(node);
+  } else {
+    delete registry[instance._onLayoutId];
+    delete instance._onLayoutId;
+  }
+};
+
 const safeOverride = (original, next) => {
   if (original) {
     return function prototypeOverride() {
-      original.call(this);
-      next.call(this);
+      /* eslint-disable prefer-rest-params */
```

Comment:
What's the reason for this?

Reply:
personally I think that it is unnecessary to use the rest operator in this case, this version should perform better than its transpiled equivalent.